netbox/core/views.py

```diff
@@ -642,11 +648,145 @@ def get(self, request):
+            'name': plugin_config.verbose_name,
+            'tag_line': plugin_config.description,
+            'description_short': plugin_config.description,
+            'author': plugin_config.author or _('Unknown Author'),
+            'created': None,
+            'updated': None,
+            'is_local': True,
+            'is_installed': True,
+            'is_certified': False,
+            'is_community': False,
+            'versions': [],
+        }
+
+    return plugins
+
+
+def get_catalog_plugins(plugins):
+    url = 'https://api.netbox.oss.netboxlabs.com/v1/plugins'
+    session = requests.Session()
+
+    def get_pages():
+        payload = {'page': '1', 'per_page': '25'}
+        first_page = session.get(url, params=payload).json()
+        yield first_page
+        num_pages = first_page['metadata']['pagination']['last_page']
+
+        for page in range(2, num_pages + 1):
+            payload['page'] = page
+            next_page = session.get(url, params=payload).json()
+            yield next_page
+
+    for page in get_pages():
+        for data in page['data']:
+
+            versions = []
+            versions.append(data['release_latest'])
+            versions.extend(data['release_recent_history'])
+            if data['slug'] in plugins:
+                plugins[data['slug']]['is_local'] = False
+                plugins[data['slug']]['is_certified'] = data['release_latest']['is_certified']
+                plugins[data['slug']]['description_short'] = data['description_short']
+            else:
+                plugins[data['slug']] = {
+                    'slug': data['slug'],
+                    'config_name': data['config_name'],
+                    'name': data['title_short'],
+                    'title_long': data['title_long'],
+                    'tag_line': data['tag_line'],
+                    'description_short': data['description_short'],
+                    'author': data['author']['name'] or _('Unknown Author'),
+                    'created': datetime_from_timestamp(data['created_at']),
+                    'updated': datetime_from_timestamp(data['updated_at']),
+                    'is_local': False,
+                    'is_installed': False,
+                    'is_certified': data['release_latest']['is_certified'],
+                    'is_community': not data['release_latest']['is_certified'],
+                    'versions': versions,
+                }
+
+    return plugins
+
+
+def get_plugins():
+    if plugins := cache.get('plugins-catalog-feed'):
+        return plugins
+
+    plugins = {}
+    plugins = get_local_plugins(plugins)
+    plugins = get_catalog_plugins(plugins)
+
+    cache.set('plugins-catalog-feed', plugins, 3600)
+    return plugins
+
+
+class PluginListView(UserPassesTestMixin, View):
+
+    def test_func(self):
+        return self.request.user.is_staff
+
+    def get(self, request):
+        q = request.GET.get('q', None)
+
+        plugins = get_plugins()
```

Comment:
The plugins list should probably show installed plugins first, followed by catalog plugins, each sorted alphabetically.

Reply:
Further asks here:
- On deployments where `settings.RELEASE.edition != 'community'`, default to showing only catalog plugins for which `release_latest.is_netboxlabs_supported == True`.
- Provide a way to toggle this behavior from the table view

One way to achieve this would be to add a query parm to this view and set it to a truthy value on non-community installations, but do what makes sense to you. Fancy client-side toggling would be cool, but only if it's easy.

If this comment needs to be a FR issue, let me know.

Reply:
@jeffgdotorg can we make the last one a FR for post beta?